CRM/Core/Payment/PayPalProIPN.php

```diff
@@ -236,16 +236,18 @@ public function recur(&$input, &$ids, &$objects, $first) {
           $recur->start_date = $now;
         }
         else {
-          $input['invoice_id'] = md5(uniqid(rand(), TRUE));
-          $input['original_contribution_id'] = $ids['contribution'];
-          $input['contribution_recur_id'] = $ids['contributionRecur'];
-
           if ($input['paymentStatus'] != 'Completed') {
             throw new CRM_Core_Exception("Ignore all IPN payments that are not completed");
           }
+
           // In future moving to create pending & then complete, but this OK for now.
           // Also consider accepting 'Failed' like other processors.
-          $input['contribution_status_id'] = 1;
+          $input['contribution_status_id'] = $contributionStatuses['Completed'];
+          $input['invoice_id'] = md5(uniqid(rand(), TRUE));
+          $input['original_contribution_id'] = $ids['contribution'];
+          $input['contribution_recur_id'] = $ids['contributionRecur'];
+          // We don't want to send out email receipts for repeat contributions.
+          $input['is_email_receipt'] = FALSE;
```

Comment:
this looks like a change in behaviour - and while I think many would prefer it I don't think a change to outgoing emails should be this PR as they need their own discussion

Reply:
Sure, I've removed that as it's not critical to this PR.